polartools/diffraction.py

```diff
@@ -0,0 +1,155 @@
+# Copyright (c) 2020, UChicago Argonne, LLC.
+# See LICENSE file for details.
+
+import numpy as np
+from pandas import DataFrame
+from lmfit.models import GaussianModel, LorentzianModel, VoigtModel, LinearModel, PseudoVoigtModel
+import matplotlib.pyplot as plt
+from .load_data import load_table
+
+def fit_peak(xdata,ydata,model='Gaussian',output=False):
+    """
+    Fit Bragg peak with a pseudo-voigt function.
+
+    Uses lmfit (https://lmfit.github.io/lmfit-py/).
+
+    WARNING: This imposes constrains to the fit that are described in
+    :func: `polartools.process_data._generate_initial_guess`
+
+    Parameters
+    ----------
+    xdata : iterable
+        List of x-axis values.
+    yydata : iterable
+        List of y-axis values.
+    model:
+        fit model: Gaussian, Lorentian, Voigt, PseudoVoigt
+    output:
+        Output fit parameters and plot data+fit for each scan.
+
+    Returns
+    -------
+    fit : lmfit ModelResult class
+        Contains the fit results. See:
+        https://lmfit.github.io/lmfit-py/model.html#the-modelresult-class
+    """
+
+    models=['Gaussian','Lorentzian','Voigt','PseudoVoigt']
+    if model in models:
+        if model=='Gaussian':
+            peak_mod=GaussianModel()
+        if model=='Lorentzian':
+            peak_mod=LorentzianModel()
+        if model=='Voigt':
+            peak_mod=VoigtModel()
+        if model=='PseudoVoigt':
+            peak_mod=PseudoVoigtModel()
+
+    background=LinearModel()
+    mod=peak_mod+background
+    pars = background.make_params(intercept=ydata.min(), slope=0)
+    pars += peak_mod.guess(ydata, x=xdata)
```

Comment:
Would it be useful to add constrains to the parameters? For instance, make sigma and amplitude > 0:
```
pars['sigma'].set(min=0)
pars['amplitude'].set(min=0)
```